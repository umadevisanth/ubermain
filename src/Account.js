import { StyleSheet, View, Image, StatusBar } from 'react-native';
import Globalstyle from './Globalstyle';
import React, { useEffect } from 'react';
import { LinearGradient } from 'expo-linear-gradient';
import { useNavigation } from '@react-navigation/native';

const Account = (props) => {
  

  const navigation = useNavigation();

  useEffect(() => {
    const timer = setTimeout(() => {
      navigation.navigate('Login');
    }, 1500); //0.5s, adjust as needed

    return () => clearTimeout(timer); // Cleanup the timer on component unmount
  }, []);

  return (
    <View style={{ flex: 1 }}>
      <StatusBar backgroundColor='#000000' />
      <LinearGradient
        colors={['#1976D2', '#535AFF']}
        style={Globalstyle.main}>
        <View style={Globalstyle.accounttextview}>
          <View style={Globalstyle.Account_view}>
            <Image
              source={require('../assets/Grouplogo.png')}
              style={Globalstyle.accountviewimage}
            />
          </View>
        </View>
      </LinearGradient>
    </View>
  );
}

export default Account;

const styles = StyleSheet.create({});
