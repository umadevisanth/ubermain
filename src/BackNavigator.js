import { View, Text, TouchableOpacity } from 'react-native'
import React from 'react'
import {Feather} from '@expo/vector-icons'
import { useNavigation } from '@react-navigation/native'
import styles from './styles'


const BackNavigator = ({titleName,navpath}) => {
    const navigation = useNavigation();
  return (
    <View style={styles.back_act_cont}>
      <TouchableOpacity onPress={()=>{
        navigation.goBack();
      }}><Feather style={styles.Back_act_cont_leftarrow} name='arrow-left' size={26}/></TouchableOpacity>
      <Text style={styles.Back_act_cont_title}>{titleName}</Text>
    </View>
  )
}

export default BackNavigator