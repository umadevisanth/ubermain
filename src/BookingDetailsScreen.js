import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet, ActivityIndicator, TouchableOpacity, StatusBar, Alert } from 'react-native';
import { firestore } from '../Config';  // Import Firestore configuration
import { useNavigation } from '@react-navigation/native';  // Import navigation hook
import { Feather } from '@expo/vector-icons';  // Import Feather icons
import MapView from 'react-native-maps';  // Import MapView component

const BookingDetailsScreen = ({ route }) => {


  const book = {

    pick:'Pickup Location:',
    drop:'Dropping Location:',
    date:'Date:',
    time:'Time:', 
    distance:'Distance:',
    amount:'Amount Per Km:',
    fare:'Fare Amount:',
    confirm:'Confirm Booking',
    km:'KM',
    inr:'INR'
    
  };
  const jsonData = JSON.stringify(book);

  const { bookingId } = route.params;  // Extract bookingId from route parameters
  const [loading, setLoading] = useState(true);  // State to manage loading indicator
  const [bookingDetails, setBookingDetails] = useState(null);  // State to store booking details

  const navigation = useNavigation();  // Get navigation object

  useEffect(() => {
    // Function to fetch booking details from Firestore
    const fetchBookingDetails = async () => {
      try {
        const documentSnapshot = await firestore.collection('bookings').doc(bookingId).get();  // Fetch booking details
        if (documentSnapshot.exists) {
          setBookingDetails(documentSnapshot.data());  // Set booking details if document exists
        } else {
          console.log('Document does not exist!');
        }
      } catch (error) {
        console.error('Error fetching booking details: ', error);  // Log any errors
      } finally {
        setLoading(false);  // Set loading to false after fetching data
      }
    };
    fetchBookingDetails();
  }, [bookingId]);  // Dependency array ensures this effect runs when bookingId changes

  // Function to handle booking confirmation
  const handleConfirmBooking = () => {
    navigation.navigate('Payment');  // Navigate to Payment screen
    Alert.alert('Success', 'Booking successfully made.');  // Show success alert
  };

  if (loading) {
    // Render loading indicator if data is still being fetched
    return (
      <View style={styles.loadingContainer}>
        <ActivityIndicator size="large" color="#0000ff" />
      </View>
    );
  }

  return (
    <View style={styles.container}>
      {/* Back button to navigate to the previous screen */}
      <TouchableOpacity style={styles.back} onPress={() => navigation.goBack()}>
        <Feather name="arrow-left" size={26} color="#979797" />
      </TouchableOpacity>
      
      {/* Display booking details */}
      <Text style={styles.distance}>{book.pick} {bookingDetails?.pickupLocation}</Text>
      <Text style={styles.distance}>{book.drop} {bookingDetails?.dropLocation}</Text>
      <Text style={styles.distance}>{book.date} {bookingDetails?.selectedDate}</Text>
      <Text style={styles.distance}>{book.time} {bookingDetails?.selectedTime}</Text>
      <Text style={styles.distance}>{book.distance} {bookingDetails?.distance}{book.km}</Text>
      <Text style={styles.distance}>{book.amount} {bookingDetails?.amountPerKm}</Text>
      <Text style={styles.distance}>{book.fare} {bookingDetails?.fareAmount} {book.inr}</Text>
      
      {/* Button to confirm the booking */}
      <TouchableOpacity
        onPress={handleConfirmBooking}
        style={styles.confirmButton}>
        <Text style={styles.confirmButtonText}>{book.confirm}</Text>
      </TouchableOpacity>
      
      {/* MapView to display location map */}
      <MapView
        style={styles.map}
        region={{
          latitude: 37.78825,
          longitude: -122.4324,
          latitudeDelta: 0.015,
          longitudeDelta: 0.0121,
        }}
      />
      <StatusBar />
    </View>
  );
};

// Styles for the component
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: '8%',
    backgroundColor: '#1A1A1A',
  },
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  distance: {
    color: '#DADADA',
    fontSize: 18,
  },
  back: {
    marginBottom: '4%',
  },
  map: {
    height: 300,
    alignItems: 'center',
  },
  confirmButton: {
    marginVertical: 30,
    width: '100%',
    backgroundColor: 'black',
    paddingHorizontal: 10,
    paddingVertical: '4%',
    fontSize: 30,
  },
  confirmButtonText: {
    textAlign: 'center',
    color: 'white',
    fontSize: 20,
  },
});

export default BookingDetailsScreen;
