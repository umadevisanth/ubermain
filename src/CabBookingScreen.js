import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, Alert } from 'react-native';
import { useRoute, useNavigation } from '@react-navigation/native';
import { Feather } from '@expo/vector-icons';
import MapView, { Marker, Polyline } from 'react-native-maps';
import Globalstyle from './Globalstyle';

const CabBookingScreen = () => {
  // Access route parameters and navigation functionalities
  const route = useRoute();
  const navigation = useNavigation();
  const { pickupLocation, dropLocation, selectedDate, selectedTime, distance, amount } = route.params;

  const AVERAGE_SPEED_KMH = 60; // Average speed in km/h

  // Function to calculate estimated travel time based on distance and average speed
  const calculateTravelTime = (distance) => {
    const timeInHours = distance / AVERAGE_SPEED_KMH;
    const hours = Math.floor(timeInHours);
    const minutes = Math.round((timeInHours - hours) * 60);
    return `${hours}h ${minutes}m`;
  };

  const travelTime = calculateTravelTime(distance);

  // Handle booking confirmation and navigation to the Payment screen
  const handleConfirmBooking = () => {
    Alert.alert('Booking Confirmed', 'Your cab booking has been confirmed.', [
      { text: 'OK', onPress: () => navigation.navigate('Payment') }
    ]);
  };

  // Get coordinates for a given location name from the list of places
  const getCoordinates = (locationName) => {
    const places = [
      { name: "Hosur", latitude: 12.743028246989406, longitude: 77.82242655968571 },
      { name: "Bangalore", latitude: 12.982049199078174, longitude: 77.59315534942606 },
      { name: "Hyderabad", latitude: 17.414901720009375, longitude: 78.47600521819076 },
      { name: "Varthur", latitude: 12.957906483800606, longitude: 77.74410601475772 },
      { name: "Salem", latitude: 11.660964499186347, longitude: 78.13954353492788 }
    ];
    return places.find(place => place.name.toLowerCase() === locationName.toLowerCase());
  };

  // Retrieve coordinates for pickup and drop locations
  const pickupCoordinates = getCoordinates(pickupLocation);
  const dropCoordinates = getCoordinates(dropLocation);

  return (
    <View style={Globalstyle.container}>
      {/* Back button to navigate to the previous screen */}
      <TouchableOpacity style={Globalstyle.back} onPress={() => navigation.goBack()}>
        <Feather name="arrow-left" size={26} color="#979797" />
      </TouchableOpacity>
      <Text style={Globalstyle.title}>Booking Details</Text>
      {/* Display booking details */}
      <View style={Globalstyle.detailContainer}>
        <Text style={Globalstyle.label}>Pickup Location:</Text>
        <Text style={Globalstyle.value}>{pickupLocation}</Text>
      </View>
      <View style={Globalstyle.detailContainer}>
        <Text style={Globalstyle.label}>Drop Location:</Text>
        <Text style={Globalstyle.value}>{dropLocation}</Text>
      </View>
      <View style={Globalstyle.detailContainer}>
        <Text style={Globalstyle.label}>Date:</Text>
        <Text style={Globalstyle.value}>{selectedDate}</Text>
      </View>
      <View style={Globalstyle.detailContainer}>
        <Text style={Globalstyle.label}>Time:</Text>
        <Text style={Globalstyle.value}>{selectedTime}</Text>
      </View>
      <View style={Globalstyle.detailContainer}>
        <Text style={Globalstyle.label}>Distance:</Text>
        <Text style={Globalstyle.value}>{distance} km</Text>
      </View>
      <View style={Globalstyle.detailContainer}>
        <Text style={Globalstyle.label}>Estimated Travel Time:</Text>
        <Text style={Globalstyle.value}>{travelTime}</Text>
      </View>
      <View style={Globalstyle.detailContainer}>
        <Text style={Globalstyle.label}>Fare:</Text>
        <Text style={Globalstyle.value}>${amount}</Text>
      </View>
      {/* Confirm booking button */}
      <TouchableOpacity style={Globalstyle.confirmButton} onPress={handleConfirmBooking}>
        <Text style={Globalstyle.confirmButtonText}>Confirm Booking</Text>
      </TouchableOpacity>
      {/* Map view to display pickup, drop locations, and route polyline */}
      {pickupCoordinates && dropCoordinates && (
        <MapView
          style={Globalstyle.map}
          initialRegion={{
            latitude: (pickupCoordinates.latitude + dropCoordinates.latitude) / 2,
            longitude: (pickupCoordinates.longitude + dropCoordinates.longitude) / 2,
            latitudeDelta: Math.abs(pickupCoordinates.latitude - dropCoordinates.latitude) + 0.1,
            longitudeDelta: Math.abs(pickupCoordinates.longitude - dropCoordinates.longitude) + 0.1,
          }}
        >
          <Marker coordinate={pickupCoordinates} title="Pickup Location" />
          <Marker coordinate={dropCoordinates} title="Drop Location" />
          {/* Polyline to draw the route between pickup and drop locations */}
          <Polyline
            coordinates={[pickupCoordinates, dropCoordinates]}
            strokeColor="#E14D2A"
            strokeWidth={2}
          />
        </MapView>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  
});

export default CabBookingScreen;
