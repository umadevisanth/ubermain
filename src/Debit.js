import React from 'react';
import { View, Text, SafeAreaView, TouchableOpacity, StyleSheet, Alert } from 'react-native';
import { useNavigation } from '@react-navigation/native';


const Debit = () => {
  const debit = {

      paytm:'Paytm Payment',
      trans:'Transaction Details:',
      amount:'Amount: $50',
      receipt:'Recipient: John Doe',
      id:'Transaction ID: 123456789',
      proceed:'Proceed to Payment',


    
  };
  const jsonData = JSON.stringify(debit);


    const navigation = useNavigation();
    const handlePayment = () => {
        // Here you would integrate with Paytm's payment gateway
        // For this example, we'll just show an alert to simulate a successful payment
        Alert.alert('Payment Successful', 'Thank you for your payment!', [
          { text: 'OK', onPress: () => navigation.navigate('Account') }, // Navigate to Home screen after payment
        ]);
      };

  return (
    <SafeAreaView style={styles.container}>
      {/* Header */}
      <View style={styles.header}>
        <Text style={styles.headerText}>{debit.paytm}</Text>
      </View>
      
      {/* Transaction Details */}
      <View style={styles.content}>
        <Text style={styles.transactionDetails}>{debit.trans}</Text>
        <Text style={styles.transactionText}>{debit.amount}</Text>
        <Text style={styles.transactionText}>{debit.receipt}</Text>
        <Text style={styles.transactionText}>{debit.id}</Text>
      </View>

      {/* Payment Button */}
      <TouchableOpacity style={styles.paymentButton} onPress={handlePayment}>
        <Text style={styles.paymentButtonText}>{debit.proceed}</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'center',
    },
    header: {
      alignItems: 'center',
      marginBottom: 20,
    },
    headerText: {
      fontSize: 24,
      fontWeight: 'bold',
    },
    content: {
      marginBottom: 30,
    },
    transactionDetails: {
      fontSize: 18,
      fontWeight: 'bold',
      marginBottom: 10,
    },
    transactionText: {
      fontSize: 16,
      marginBottom: 5,
    },
    paymentButton: {
      backgroundColor: 'blue',
      paddingVertical: 10,
      paddingHorizontal: 20,
      borderRadius: 5,
    },
    paymentButtonText: {
      color: '#fff',
      fontSize: 18,
      fontWeight: 'bold',
    },
  });

export default Debit