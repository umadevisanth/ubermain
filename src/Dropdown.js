import React, { useState } from 'react';
import { View, TouchableOpacity, Modal, Image, Text } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome'; // or any other icon library you prefer

const Dropdown = ({ onSelectItem }) => {
  const [showDropdown, setShowDropdown] = useState(false);
  const [selectedCountryCode, setSelectedCountryCode] = useState('');

  const toggleDropdown = () => {
    setShowDropdown(!showDropdown);
  };

  const handleItemSelect = (countryCode, image) => {
    setSelectedCountryCode(countryCode);
    onSelectItem({ image, countryCode }); // Pass both image and country code back to parent
    toggleDropdown(); // Close dropdown after item selection
  };

  return (
    <View>
      <TouchableOpacity onPress={toggleDropdown}>
        <Icon name="caret-down" size={24} color="#EDF6FF" />
      </TouchableOpacity>

      <Modal visible={showDropdown} animationType="slide">
        <View style={{ justifyContent: 'center', alignItems: 'center' }}>
          <TouchableOpacity onPress={() => handleItemSelect('+91', require('../assets/Flag.jpeg'))}>
            <Image style={{ height: 30, width: 40 }} source={require('../assets/Flag.jpeg')} />
            <Text>+91</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => handleItemSelect('+1', require('../assets/US.jpeg'))}>
            <Image style={{ height: 30, width: 40 }} source={require('../assets/US.jpeg')} />
            <Text>+1</Text>
          </TouchableOpacity>
          {/* Add more dropdown items as needed */}
          <TouchableOpacity onPress={toggleDropdown}>
            <Text>Close Dropdown</Text>
          </TouchableOpacity>
        </View>
      </Modal>
      {/* Display the selected country code */}
      {selectedCountryCode !== '' && <Text>Selected Country Code: {selectedCountryCode}</Text>}
    </View>
  );
};

export default Dropdown;
