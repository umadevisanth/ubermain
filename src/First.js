import React, { useState } from 'react';
import { View, Text, SafeAreaView, TouchableOpacity, TextInput } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import Globalstyle from './Globalstyle';
import { Feather } from '@expo/vector-icons';


const First = () => {
  const navigation = useNavigation();
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [firstNameError, setFirstNameError] = useState('');
  const [lastNameError, setLastNameError] = useState('');
  const first = {
    name: 'What`s Your Name?',
    next: 'Next'
  };

  const handleNextPress = () => {
    const minNameLength = 2; // Minimum length for name fields

    // Check if first name is empty or does not meet minimum length
    if (!firstName || firstName.length < minNameLength) {
      setFirstNameError('First name is required .');
      return;
    }

    // Check if last name is empty or does not meet minimum length
    if (!lastName || lastName.length < minNameLength) {
      setLastNameError('Last name is required .');
      return;
    }

    // Check if first name contains numeric characters
    if (/\d/.test(firstName)) {
      setFirstNameError('First name cannot contain numeric characters.');
      return;
    }

    // Check if last name contains numeric characters
    if (/\d/.test(lastName)) {
      setLastNameError('Last name cannot contain numeric characters.');
      return;
    }

    // Reset errors if validation passes
    setFirstNameError('');
    setLastNameError('');

    // Navigate to the next screen if validation passes
    navigation.navigate('Profile');
  };

  return (
    <SafeAreaView style={Globalstyle.firstsafeview}>
      <View style={Globalstyle.firstviewtext}>
        <TouchableOpacity onPress={() => navigation.goBack('Social')}>
          <Feather name="arrow-left" size={24} color="#EDF6FF" style={Globalstyle.First_left} />

        </TouchableOpacity>
      </View>
      <View style={{ flex: 5 }}>
        <View style={Globalstyle.firstviewtexttext}>
          <Text style={Globalstyle.First_name}>{first.name}</Text>
        </View>

        <View style={Globalstyle.First_view}>
          <View >
            <TextInput
              style={{ color: '#EDF6FF', width: '35%', fontSize: 18, borderBottomWidth: 1, borderColor: firstNameError ? 'red' : '#EDF6FF', padding: 20, width: 150 }}
              placeholder="First" placeholderTextColor={'#EDF6FF'}
              value={firstName}
              onChangeText={(text) => { setFirstName(text.replace(/[^A-Za-z]/g, '')); setFirstNameError(''); }}
            />
            {firstNameError ? <Text style={Globalstyle.firstError}>{firstNameError}</Text> : null}
          </View>
          <View>
            <TextInput
              style={{ color: '#EDF6FF', width: '35%', fontSize: 18, borderBottomWidth: 1, borderColor: lastNameError ? 'red' : '#EDF6FF', padding: 20, width: 150 }}
              placeholder="Last" placeholderTextColor={'#EDF6FF'}
              value={lastName}
              onChangeText={(text) => { setLastName(text.replace(/[^A-Za-z]/g, '')); setLastNameError(''); }}
            />
            {lastNameError ? <Text style={Globalstyle.lastError}>{lastNameError}</Text> : null}
          </View>
        </View>
      </View>
      <View style={{ flex: 1 }}>
        <TouchableOpacity
          style={Globalstyle.First_next}
          onPress={handleNextPress}
        >
          <Text style={Globalstyle.First_nexttext}>{first.next}</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

export default First;