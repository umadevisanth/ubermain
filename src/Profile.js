import React, { useState } from 'react';
import { View, Text, SafeAreaView, TouchableOpacity, Image, Linking } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import CheckBox from 'expo-checkbox';
import Globalstyle from './Globalstyle';
import { Feather } from '@expo/vector-icons';

const Profile = () => {
  const profile = {
    tapping:
      'By tapping the arrow below, you agree to Uberâ€™s Terms of Use and acknowledge that you have read the Privacy Policy.',
    check:
      'Check the box to indicate that you are at least 18 years of age, agree to the Terms & Conditions and acknowledge the Privacy Policy.',
    next: 'Next',
    checkbox:'Check the box to indicate that you are at least 18 years of age, agree to the',
    terms:'Terms & Conditions',
    ack:'and acknowledge the',
    policy:'Privacy Policy'
  
  };

  const navigation = useNavigation(); // Hook to manage navigation
  const [isChecked, setIsChecked] = useState(false); // State to manage checkbox status

  const handleNextPress = () => {
    if (isChecked) {
      navigation.navigate('Payment'); // Navigate to Payment screen if checkbox is checked
    }
  };

  return (
    <SafeAreaView style={Globalstyle.profilesafetextview}>
      <View>
        {/* Back button to navigate to the previous screen */}
        <TouchableOpacity onPress={() => navigation.navigate('First')}>
          <Feather name="arrow-left" size={24} color="#EDF6FF" style={Globalstyle.Profile_lefticon} />
        </TouchableOpacity>
      </View>

      {/* Profile image section */}
      <View style={Globalstyle.Profile_profileimage}>
        <View style={Globalstyle.whole_image}>
          <Image style={Globalstyle.ellipse} source={require('../assets/Ellipse1.png')} />
        </View>
        <Image style={Globalstyle.icon_profile} source={require('../assets/vs_profile.png')} />
      </View>

      {/* Text describing terms and privacy policy */}
      <View>
        <Text style={Globalstyle.Profile_tapping}>{profile.tapping}</Text>
      </View>

      {/* Checkbox and terms text */}
      <View>
        <View style={Globalstyle.Profile_text}>
          <CheckBox value={isChecked} onValueChange={setIsChecked} />
          <Text style={Globalstyle.Profile_check}>
            {profile.checkbox}{' '}
            <Text
              style={{ color: 'blue' }}
              onPress={() => Linking.openURL('https://www.uber.com/legal/terms/us/')}
            >
              {profile.terms}
            </Text>{' '}
            {profile.ack}{' '}
            <Text
              style={{ color: 'blue' }}
              onPress={() => Linking.openURL('https://www.uber.com/legal/privacy/us/')}
            >
              {profile.policy}
            </Text>
            .
          </Text>
        </View>

        {/* Next button displayed only if checkbox is checked */}
        {isChecked && (
          <TouchableOpacity style={Globalstyle.Profile_checked} onPress={handleNextPress}>
            <Text style={Globalstyle.Profile_next}>{profile.next}</Text>
          </TouchableOpacity>
        )}
      </View>
    </SafeAreaView>
  );
};

export default Profile;
