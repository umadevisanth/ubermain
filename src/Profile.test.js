import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import {Profile} from '../src/Profile';
import CheckBox from 'expo-checkbox';

// Mock @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
    AntDesign: 'AntDesign', // Mock the specific icon component you're using
}));

// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

import { View } from 'react-native';
jest.mock('expo-checkbox', () => ({
  CheckBox: jest.fn(() => ({
    setNativeProps: jest.fn(),
  })),
}));


describe('Profile component', () => {
  it('renders correctly', () => {
    // Mock useNavigation behavior
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    // Render the Login component
    const tree = renderer.create(<Profile />).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});
