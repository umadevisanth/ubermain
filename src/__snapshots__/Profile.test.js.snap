// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "black",
      "height": "100%",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <AntDesign
        color="black"
        name="left"
        size={20}
        style={
          {
            "color": "#979797",
            "height": 16,
            "marginLeft": 30,
            "marginTop": 40,
            "width": 16,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 100,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/Profile.png",
        }
      }
      style={
        {
          "height": 130,
          "width": 130,
        }
      }
    />
  </View>
  <View
    style={
      {
        "marginTop": 60,
      }
    }
  >
    <Text
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 20,
          "lineHeight": 23.14,
          "marginLeft": "auto",
          "marginRight": "auto",
          "width": "85%",
        }
      }
    >
      By tapping the arrow below, you agree to Uberâ€™s Terms of Use and acknowledge that you have read the Privacy Policy.
    </Text>
  </View>
  <View
    style={
      {
        "marginTop": 90,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "width": "85%",
        }
      }
    >
      <View
        testID="mock-checkbox"
      />
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 16,
            "lineHeight": 17.58,
            "width": "90%",
          }
        }
      >
        Check the box to indicate that you are at least 18 years of age, agree to the Terms & Conditions and acknowledge the Privacy Policy.
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
